### PR TITLE
Wire Percentiles aggregator into new VS framework

### PR DESCRIPTION
**Pull request by polyfractal**
This required a bit of a refactor to percentiles itself.  Before, the Builder would switch on the chosen algo to generate an algo-specific factory.  This doesn't work (or at least, would be difficult) in the new VS framework.

This refactor consolidates both factories together and introduces a PercentilesConfig object to act as a standardized way to pass algo-specific parameters through the factory.  This object is then used when deciding which kind of aggregator to create

Note: `CoreValuesSourceType.HISTOGRAM` still lives in core, and will be moved in a subsequent PR.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesAggregatorFactory.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                        return new TDigestPercentilesAggregator(name, valuesSource, context, parent, percents, compression, keyed,
+                            formatter, pipelineAggregators, metaData);
+                    } else if (percentilesConfig.getMethod().equals(PercentilesMethod.HDR)) {
+                        int numSigFig = ((PercentilesConfig.HdrHistoConfig)percentilesConfig).getNumberOfSignificantValueDigits();
+                        return new HDRPercentilesAggregator(name, valuesSource, context, parent, percents, numSigFig, keyed,
+                            formatter, pipelineAggregators, metaData);
+                    }
+
+                    // This should already have thrown but just in case
+                    throw new IllegalArgumentException("Unknown percentiles method: [" + percentilesConfig.getMethod().toString() + "]");
```
Comment (not-napoleon) [inline server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesAggregatorFactory.java]: In the builder, we throw `IllegalStateException` for an unknown method.  Even if we never expect to hit this, we should be consistent.  I don't know which is correct, but we should pick one and use it.

**Review by not-napoleon — APPROVED** [review]
+1, LGTM

File: server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesAggregatorFactory.java (hunk 1 above)
Comment (polyfractal) [reply server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesAggregatorFactory.java]: :+1: good point, hadn't noticed.  Probably stick to what the builder is throwing (at least for now) so it's consistent with how it works today.

**Comment by polyfractal** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix-unix

**Pull request merged**